images-and-plotting-instructor.ipynb

```diff
@@ -0,0 +1,718 @@
+   "source": [
+    "## Arrays as images"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All photographic images represent a measurement of how much light hits the receiver. For instance, the Hubble image below is obtained by measuring the brightnesses of distant stars:\n",
+    "\n",
+    "![Hubble image](fig/galaxy.jpg)\n",
+    "\n",
+    "With traditional optical cameras, this measurement results in an image which is continuous, as it is projected directly onto paper. In order to store images digitally, they need to be divided into discrete chunks, pixels, each of which contains the value of the measurement in that small portion of the image. In this representation, an image is simply a grid of numbers, which allows it to be easily stored as an array with a shape equal to the resolution of the image.\n",
+    "\n",
+    "The `scikit-image` (abbreviated to `skimage` in code) module contains some sample images in the `data` submodule that we can use to demonstrate this principle."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Get some import statements out of the way.\n",
+    "from __future__ import division, print_function\n",
+    "%matplotlib inline\n",
+    "import matplotlib.pyplot as plt\n",
+    "from skimage import data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Load the data.moon() image and print it\n",
+    "moon = data.moon()\n",
+    "print(moon)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once read in as an array, the image can be processed in the same ways as any other array. For instance, we can easily find the highest, lowest and mean values of the image, the type of the variables stored in the array, and the resolution of the image:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Output the image minimum, mean and maximum.\n",
+    "print('Image min:', moon.min(), '; Image mean:', moon.mean(), '; Image max: ', moon.max())\n",
+    "\n",
+    "# Output the array dtype.\n",
+    "print('Data type:', moon.dtype)\n",
+    "\n",
+    "# Output image resolution.\n",
+    "print('Image resolution:', moon.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tells us that the image has a resolution of 512 x 512 pixels, and is stored as integers between 0 and 255. This is a common way of normalising images, but they can just as easily be stored as floats between 0 and 1. More commonly with astronomical data though, an image will consist of photon counts (i.e. integers), so the minimum will be 0 and there may be no upper limit to possible values. "
```

Comment:
I believe every CCD has an upper value, one due to saturation, and the other due to the digitalisation of the image, 8bit, 16,... etc.